src/assets/scripts/client/airline/AirlineModel.js

```diff
@@ -1,293 +1,200 @@
 import _get from 'lodash/get';
-import _has from 'lodash/has';
+import _head from 'lodash/head';
 import _forEach from 'lodash/forEach';
-import { choose, choose_weight } from '../utilities/generalUtilities';
+import _isArray from 'lodash/isArray';
+import _isEmpty from 'lodash/isEmpty';
+import _isObject from 'lodash/isObject';
+import _map from 'lodash/map';
+import _random from 'lodash/random';
+import _uniq from 'lodash/uniq';
+import BaseModel from '../base/BaseModel';
 
 /**
  * An aircrcraft operating agency
  *
  * @class AirlineModel
+ * @extends BaseModel
  */
-export default class AirlineModel {
+export default class AirlineModel extends BaseModel {
     /**
      * Create new airline
      *
      * @constructor
      * @for AirlineModel
-     * @param icao {string}
-     * @param options {object}
+     * @param airlineDefinition {object}
      */
-    constructor(icao, options) {
-        // ICAO airline designation
-        this.icao = icao;
-        // Agency name
-        this.name = _get(options, 'name', 'Default airline');
-        // Radio callsign
+    /* istanbul ignore next */
+    constructor(airlineDefinition) {
+        super(airlineDefinition);
+
+        if (!_isObject(airlineDefinition) || _isArray(airlineDefinition) || _isEmpty(airlineDefinition)) {
+            // eslint-disable-next-line max-len
+            throw new TypeError(`Invalid airlineDefinition received by AirlineModel. Expected an object but received ${typeof airlineDefinition}`);
+        }
+
+        /**
+         * ICAO airline designation
+         *
+         * @property icao
+         * @type {string}
+         */
+        this.icao = airlineDefinition.icao;
+
+        /**
+         * Agency name
+         *
+         * @property name
+         * @type {string}
+         */
+        this.name = _get(airlineDefinition, 'name', 'Default airline');
+
+        /**
+         * Radio callsign
+         *
+         * @property callsign
+         * @type {string}
+         * @default 'Default'
+         */
         this.callsign = 'Default';
-        // Parameters for flight number generation
+
+        /**
+         * Parameters for flight number generation
+         *
+         * @property flightNumberGeneration
+         * @type {Object}
+         */
         this.flightNumberGeneration = {
-            // How many characters in the flight number
+            /**
+             * Length of callsign
+             *
+             * @memberof flightNumberGeneration
+             * @property length
+             * @type {number}
+             * @default 3
+             */
             length: 3,
-            // Whether to use alphabetical characters
+
+            /**
+             * Whether to use alphabetical characters
+             *
+             * @memberof flightNumberGeneration
+             * @property alpha
```

Comment:
pulled this straight from the airline json and I'd like this to mirror that same definition.